geniusyield-server-lib/src/GeniusYield/Server/ErrorMiddleware.hs

```diff
@@ -135,6 +100,39 @@ exceptionHandler =
       WH $ \case
         GYConversionException convErr → someBackendError $ tShow convErr
         GYQueryUTxOException txErr → someBackendError $ tShow txErr
+        e@(GYBuildTxException buildErr) → case buildErr of
```

Comment:
It would be nice if the special cases here could be covered by tests.

I don't see any test changes in this PR. I would expect to see some updated tests when such logic changes.

Reply:
It is largely governed by type system. There is no longer `BuildTxException`, IIRC it's renamed to GYBuildTxError which no longer has `Exception` instance so it cannot be thrown in first place. This `GYBuildTxError` is now part of  `GYBuildTxException` and so there has been reordering of these error messages.